### PR TITLE
Update to .NET 9

### PR DESCRIPTION
**Pull request by captainsafia**
- Upgrade eShop to .NET 9 RC 1 SDK
- Update eShop to Aspire.Hosting 8.2.0
- Update TFMs on all projects to `net9.0`
- Update eShop to use new Microsoft.AspNetCore.OpenApi implementation

### CODE REVIEW

**Review by adityamandaleeka — APPROVED** [review]

File: src/eShop.ServiceDefaults/OpenApi.Extensions.cs [hunk 1]
```diff
@@ -21,45 +22,13 @@ public static IApplicationBuilder UseDefaultOpenApi(this WebApplication app)
...
-                if (authSection.Exists())
-                {
-                    setup.OAuthClientId(authSection.GetRequiredValue("ClientId"));
-                    setup.OAuthAppName(authSection.GetRequiredValue("AppName"));
-                }
-            });
-
-            // Add a redirect from the root of the app to the swagger endpoint
-            app.MapGet("/", () => Results.Redirect("/swagger")).ExcludeFromDescription();
+            var authSection = openApiSection.GetSection("Auth");
```
Comment (BrennanConroy) [inline src/eShop.ServiceDefaults/OpenApi.Extensions.cs:29]: Remove?

File: src/eShop.ServiceDefaults/OpenApi.Extensions.cs [hunk 2]
```diff
@@ -69,26 +38,50 @@ public static IHostApplicationBuilder AddDefaultOpenApi(
...
+                    options.ApplySecuritySchemeDefinitions();
+                    options.ApplyOperationDeprecatedStatus();
+                });
+            }
         }
 
         return builder;
     }
+
+    internal static T GetService<T>(this IServiceCollection services)
```
Comment (BrennanConroy) [inline src/eShop.ServiceDefaults/OpenApi.Extensions.cs:75]: What's this for?

**Review by BrennanConroy — APPROVED** [review]

File: src/eShop.ServiceDefaults/OpenApi.Extensions.cs (hunk 1 above)
Comment (captainsafia) [reply src/eShop.ServiceDefaults/OpenApi.Extensions.cs:29]: Yeah, we can remove this for now.

For context, Scalar hasn't shipped support for configuring OAuth through an options object so we can't actually use these configuration options yet.

File: src/eShop.ServiceDefaults/OpenApi.Extensions.cs (hunk 2 above)
Comment (captainsafia) [reply src/eShop.ServiceDefaults/OpenApi.Extensions.cs:75]: We can delete this. It was from an experiment try to get a nicer integration with API versioning working.

**Pull request merged**